src/environment.ts

```diff
@@ -80,14 +82,14 @@ export class SecureEnvironment {
     // raw object map
     private rom: WeakMap<RawFunction | RawObject, SecureRecord> = WeakMapCreate();
     // distortion mechanism (default to noop)
-    private distortionCallback?: (target: SecureProxyTarget) => SecureProxyTarget = t => t;
+    private distortionMap: DistortionMap;
 
     constructor(options: SecureEnvironmentOptions) {
         if (isUndefined(options)) {
             throw ErrorCreate(`Missing SecureEnvironmentOptions options bag.`);
         }
-        const { rawGlobalThis, secureGlobalThis, distortionCallback } = options;
-        this.distortionCallback = distortionCallback;
+        const { rawGlobalThis, secureGlobalThis, distortionMap } = options;
```

Comment:
plucking properties off of an options object can be influenced by `Object.prototype` additions without `HasOwnProperty` checks.

Reply:
Since this is constructor, I feel that the API is a regular options object, we should not impose restrictions on what they pass it, maybe they create a class to configure the options, etc. I believe the important bits are how we ingest them, and normalize them on the spot.

basically, IMO, public APIs should not impose this restrictions on arguments passed to them, just normalize them to protect usage of them later on.